barman/compression.py

```diff
@@ -603,6 +611,36 @@ def validate(self, pg_server_version, remote_status):
         return issues
 
 
+class ZSTDPgBaseBackupCompressionOption(PgBaseBackupCompressionOption):
+    def validate(self, pg_server_version, remote_status):
+        """
+        Validate zstd-specific options.
+
+        :param pg_server_version int: the server for which the
+          compression options should be validated.
+        :param dict remote_status: the status of the pg_basebackup command
+        :return List: List of Issues (str) or empty list
+        """
+        issues = super(ZSTDPgBaseBackupCompressionOption, self).validate(
+            pg_server_version, remote_status
+        )
+        # "backup_location = server" requires pg_basebackup >= 15
+        if remote_status["pg_basebackup_version"] < Version("15"):
+            issues.append(
+                "backup_compression = %s requires "
+                "pg_basebackup 15 or greater" % self.config.type
+            )
+
+        if self.config.level is not None and (
+            self.config.level < 1 or self.config.level > 22
```

Comment:
Negative values are allowed here since the zstd library supports [negative compression levels down to -99](https://github.com/facebook/zstd/blob/03cc84fddb9747ddc63f76b6f3171cd8f66da202/programs/zstdcli.c#L796) and pg_basebackup will [pass whatever level is provided to the zstd library](https://github.com/postgres/postgres/blob/REL_15_BETA3/src/backend/replication/basebackup_zstd.c#L99-L106) (unless it's `0`).

Reply:
Ok - that makes sense.